diff --git a/Resources/Config/descriptions.plist b/Resources/Config/descriptions.plist
index cdf2b8c..b198a6a 100644
--- a/Resources/Config/descriptions.plist
+++ b/Resources/Config/descriptions.plist
@@ -1767,7 +1767,7 @@
 	"loadsavescreen-commander-name" = "Commander";
 	"loadsavescreen-rating"			= "Rating";
 	"loadsavescreen-no-commanders-found" = "No commanders found";
-	"loadsavescreen-commander-@-rated-@-has-@-legal-status-@-ship-@-location-@-g-@-timestamp-@" = "Commander %@ is rated %@ and has %@ in the bank. Legal status: %@. Ship: %@. Location: %@ (G%d). Timestamp: %@";
+	"loadsavescreen-commander-@-rated-@-has-@-legal-status-@-ship-@-location-@-g-@-eco-@-gov-@-tl-@-timestamp-@" = "Commander %@ is rated %@ and has %@ in the bank. Legal status: %@. Ship: %@. Location: %@ (G%d%@%@%@). Timestamp: %@";
 	"loadsavescreen-hold-@-and-press-return-to-open-folder-@"= "Hold %@ and press return to open folder: %@";
 	"loadsavescreen-hold-@-and-press-return-to-open-parent-folder-@" = "Hold %@ and press return to open parent folder: %@";
 	
diff --git a/src/Core/Entities/PlayerEntity.m b/src/Core/Entities/PlayerEntity.m
index 454c53c..5b3b64a 100644
--- a/src/Core/Entities/PlayerEntity.m
+++ b/src/Core/Entities/PlayerEntity.m
@@ -883,6 +883,12 @@ NSComparisonResult marketSorterByMassUnit(id a, id b, void *market);
 	if (![UNIVERSE inInterstellarSpace])
 	{
 		[result setObject:[UNIVERSE getSystemName:[self currentSystemID]] forKey:@"current_system_name"];
+		OOGovernmentID government = [[UNIVERSE currentSystemData] oo_intForKey:KEY_GOVERNMENT];
+		OOTechLevelID techlevel = [[UNIVERSE currentSystemData] oo_intForKey:KEY_TECHLEVEL];
+		OOEconomyID economy = [[UNIVERSE currentSystemData] oo_intForKey:KEY_ECONOMY];
+		[result setObject:[NSNumber numberWithUnsignedShort:government] forKey:@"current_system_government"];
+		[result setObject:[NSNumber numberWithUnsignedInt:techlevel] forKey:@"current_system_techlevel"];
+		[result setObject:[NSNumber numberWithUnsignedShort:economy] forKey:@"current_system_economy"];
 	}
 	
 	[result setObject:[self commanderName] forKey:@"player_name"];
diff --git a/src/Core/Entities/PlayerEntityLoadSave.m b/src/Core/Entities/PlayerEntityLoadSave.m
index 29f3288..e89a3b5 100644
--- a/src/Core/Entities/PlayerEntityLoadSave.m
+++ b/src/Core/Entities/PlayerEntityLoadSave.m
@@ -1276,7 +1276,7 @@ NSComparisonResult sortCommanders(id cdr1, id cdr2, void *context)
 	int			galNumber;
 	NSString		*timeStamp  = nil;
 	NSString 		*locationName = [cdr oo_stringForKey:@"current_system_name"];
-	
+
 	// If there is no key containing the name of the current system in
 	// the savefile, calculating what it should have been is going to
 	// be tricky now that system generation isn't seed based - but
@@ -1288,6 +1288,16 @@ NSComparisonResult sortCommanders(id cdr1, id cdr2, void *context)
 	}
 	
 	galNumber = [cdr oo_intForKey:@"galaxy_number"] + 1;	// Galaxy numbering starts at 0.
+
+	NSString *locationGov = @"";
+	NSString *locationEco = @"";
+	NSString *locationTL = [cdr objectForKey:@"current_system_techlevel"] ? [NSString stringWithFormat:@"%u", [cdr oo_unsignedIntForKey:@"current_system_techlevel"] + 1] : nil;
+	if (locationTL)
+	{
+		locationGov = [NSString stringWithFormat:@"%u", [cdr oo_unsignedShortForKey:@"current_system_government"]];
+		locationEco = [NSString stringWithFormat:@" %u", [cdr oo_unsignedShortForKey:@"current_system_economy"]];
+	}
+	else  locationTL = @"";
 	
 	timeStamp = ClockToString([cdr oo_doubleForKey:@"ship_clock" defaultValue:PLAYER_SHIP_CLOCK_START], NO);
 	
@@ -1295,7 +1305,7 @@ NSComparisonResult sortCommanders(id cdr1, id cdr2, void *context)
 	
 	NSString		*cdrDesc = nil;
 	
-	cdrDesc = [NSString stringWithFormat:DESC(@"loadsavescreen-commander-@-rated-@-has-@-legal-status-@-ship-@-location-@-g-@-timestamp-@"),
+	cdrDesc = [NSString stringWithFormat:DESC(@"loadsavescreen-commander-@-rated-@-has-@-legal-status-@-ship-@-location-@-g-@-eco-@-gov-@-tl-@-timestamp-@"),
 		[cdr oo_stringForKey:@"player_name"],
 		rating,
 		OOCredits(money),
@@ -1303,8 +1313,13 @@ NSComparisonResult sortCommanders(id cdr1, id cdr2, void *context)
 		shipName,
 		locationName,
 		galNumber,
+		locationEco,
+		locationGov,
+		locationTL,
 		timeStamp];
 	
+	//-------------------------------------------------------------------------------------------------------------------------
+	
 	[gui addLongText:cdrDesc startingAtRow:CDRDESCROW align:GUI_ALIGN_LEFT];
 	
 }
