Index: GNUmakefile
===================================================================
--- GNUmakefile	(revision 1831)
+++ GNUmakefile	(working copy)
@@ -11,18 +11,18 @@
 	ADDITIONAL_CFLAGS = -DWIN32 -DDOCKING_CLEARANCE_ENABLED -DNEED_STRLCPY `sdl-config --cflags`
 # note the vpath stuff above isn't working for me, so adding src/SDL and src/Core explicitly
 	ADDITIONAL_OBJCFLAGS = -DLOADSAVEGUI -DWIN32 -DXP_WIN -DDOCKING_CLEARANCE_ENABLED -Wno-import `sdl-config --cflags`
-ifeq ($(BUILD_WITH_DEBUG_FUNCTIONALITY),no)
-	ADDITIONAL_CFLAGS += -DNDEBUG
-	ADDITIONAL_OBJCFLAGS += -DNDEBUG
-endif
 	oolite_LIB_DIRS += -L/usr/local/lib -L$(GNUSTEP_LOCAL_ROOT)/lib -Ldeps/Windows-x86-deps/lib
 else
 	ADDITIONAL_INCLUDE_DIRS = -Ideps/Cross-platform-deps/SpiderMonkey/js/src  -Ideps/Cross-platform-deps/SpiderMonkey/js/src/Linux_All_DBG.OBJ -Isrc/SDL -Isrc/Core -Isrc/BSDCompat -Isrc/Core/Scripting -Isrc/Core/Materials -Isrc/Core/Entities -Isrc/Core/OXPVerifier -Isrc/Core/Debug
-	ADDITIONAL_OBJC_LIBS = -lpng -ljs -lGLU -lGL -lSDL -lpthread -lSDL_mixer -lgnustep-base
+	ADDITIONAL_OBJC_LIBS = -lpng deps/Cross-platform-deps/SpiderMonkey/js/src/Linux_All_DBG.OBJ/libjs.a -lGLU -lGL -lSDL -lSDL_mixer -lgnustep-base
 	ADDITIONAL_CFLAGS = -DLINUX -DDOCKING_CLEARANCE_ENABLED -DNEED_STRLCPY `sdl-config --cflags`
 	ADDITIONAL_OBJCFLAGS = -std=c99 -DLOADSAVEGUI -DLINUX -DDOCKING_CLEARANCE_ENABLED -DXP_UNIX -Wno-import `sdl-config --cflags`
 	oolite_LIB_DIRS += -Ldeps/Cross-platform-deps/SpiderMonkey/js/src/Linux_All_DBG.OBJ -L/usr/X11R6/lib/
 endif
+ifeq ($(BUILD_WITH_DEBUG_FUNCTIONALITY),no)
+	ADDITIONAL_CFLAGS += -DNDEBUG
+	ADDITIONAL_OBJCFLAGS += -DNDEBUG
+endif
 OBJC_PROGRAM_NAME = oolite
 
 oolite_C_FILES = legacy_random.c strlcpy.c OOTCPStreamDecoder.c
