Index: src/Core/Scripting/OOJSSystem.m
===================================================================
--- src/Core/Scripting/OOJSSystem.m	(revision 1827)
+++ src/Core/Scripting/OOJSSystem.m	(working copy)
@@ -50,7 +50,7 @@
 static BOOL GetRelativeToAndRange(JSContext *context, uintN *ioArgc, jsval **ioArgv, Entity **outRelativeTo, double *outRange) NONNULL_FUNC;
 static NSArray *FindJSVisibleEntities(EntityFilterPredicate predicate, void *parameter, Entity *relativeTo, double range);
 static NSArray *FindShips(EntityFilterPredicate predicate, void *parameter, Entity *relativeTo, double range);
-static OOInteger CompareEntitiesByDistance(id a, id b, void *relativeTo);
+static NSComparisonResult CompareEntitiesByDistance(id a, id b, void *relativeTo);
 
 
 static JSBool SystemGetProperty(JSContext *context, JSObject *this, jsval name, jsval *outValue);
@@ -1001,7 +1001,7 @@
 }
 
 
-static OOInteger CompareEntitiesByDistance(id a, id b, void *relativeTo)
+static NSComparisonResult CompareEntitiesByDistance(id a, id b, void *relativeTo)
 {
 	Entity				*ea = a,
 	*eb = b,

